### PR TITLE
test(mcp): add tests for portfolio/position/sentiment tools + middleware & resources

### LINKED ISSUE #124: test(mcp): add tests for portfolio/position/sentiment tools + middleware & resources
## Goal
ポートフォリオ/ポジション/センチメント系 MCP ツールと security 隣接モジュールのテストを追加。

## Background
テスト無し・0%: `portfolio_analytics.py`, `position_history.py`, `etf_calculator_tools.py`, `sentiment_analysis.py`, `ib_sec_mcp/tools/etf_calculator.py`(26%), `mcp/middleware.py`(25%, error masking/security), `mcp/resources.py`(16%)。
sentiment analyzer も欠落: `analyzers/sentiment/options.py`, `technical.py`（テスト無し）。

## Tasks
- [ ] `tests/mcp/test_portfolio_analytics.py`（sharpe/beta/correlation、価格モック）
- [ ] `tests/mcp/test_position_history.py`（SQLite フィクスチャ）
- [ ] `tests/mcp/test_etf_calculator_tools.py` ＋ `tests/tools/test_etf_calculator.py`（純ロジック）
- [ ] `tests/mcp/test_sentiment_analysis.py`
- [ ] `tests/analyzers/sentiment/test_technical.py`（※ options.py のデッド分岐削除は #CLEANUP に依存。削除後の挙動でテスト）
- [ ] `tests/mcp/test_middleware.py`（エラーマスキング/機密漏れなし）
- [ ] `tests/mcp/test_resources.py`（9 resource の内容）

## Acceptance
- 対象モジュールの coverage 向上、middleware の機密マスキングを明示検証、グリーン

## Parallel-safety
Owns: `tests/mcp/test_{portfolio_analytics,position_history,etf_calculator_tools,sentiment_analysis,middleware,resources}.py`, `tests/tools/**`, `tests/analyzers/sentiment/test_technical.py`
Wave 3。`sentiment/options.py` 本体は触らない（#CLEANUP 所有）。

### PR DESCRIPTION
**Pull request by knishioka**
## Summary

Resolves #124 (Wave 3, `test` / `parallel-safe`).

Adds unit tests for MCP tools and security-adjacent modules that previously had **0% or minimal** coverage. No production source was modified — purely test additions within the files this issue owns.

## New test files

| Test file | Target module | What it covers |
|---|---|---|
| `tests/mcp/test_portfolio_analytics.py` | `portfolio_analytics.py` | sharpe/beta/correlation happy path (yfinance mocked) + validation/early-exit branches |
| `tests/mcp/test_position_history.py` | `position_history.py` | all 5 tools against a real `PositionStore` SQLite fixture + invalid-date handling |
| `tests/mcp/test_etf_calculator_tools.py` | `etf_calculator_tools.py` | swap/portfolio-swap wrappers, trading-fee env resolution, payback null serialization |
| `tests/tools/test_etf_calculator.py` | `tools/etf_calculator.py` | pure Decimal swap math, infinite-payback edge case, `validate_etf_price` |
| `tests/mcp/test_sentiment_analysis.py` | `sentiment_analysis.py` | source routing, interpretation thresholds, timeout → `IBTimeoutError`, error masking |
| `tests/analyzers/sentiment/test_technical.py` | `analyzers/sentiment/technical.py` | RSI/MACD helpers + trend branches + graceful degradation |
| `tests/mcp/test_middleware.py` | `mcp/middleware.py` | error tracking, retry exponential backoff, **no-secret-leak verification** |
| `tests/mcp/test_resources.py` | `mcp/resources.py` | content of the 9 resources + no-data error branches |

## Coverage gains

| Module | Before | After |
|---|---|---|
| portfolio_analytics.py | 0% | 81% |
| position_history.py | 0% | 61% |
| etf_calculator_tools.py | 0% | 100% |
| sentiment_analysis.py | 0% | 87% |
| tools/etf_calculator.py | 26% | 97% |
| middleware.py | 25% | 97% |
| resources.py | 16% | 78% |
| sentiment/technical.py | 0% | 85% |

## Acceptance criteria

- [x] Coverage of target modules increased
- [x] Middleware confidential-data masking explicitly verified (logging middleware does not leak request payloads/secrets)
- [x] Green — full suite **1068 passed**

## Notes

- `sentiment/options.py` source is intentionally **not** touched (owned by #CLEANUP); per the issue, only `test_technical.py` is added under `tests/analyzers/sentiment/`.
- Tests follow existing conventions (`_fastmcp_helpers.call_tool_fn`, real storage classes via `tmp_path`, yfinance patched at module level, Decimal-based financial assertions).

## Testing

```
uv run pytest        # 1068 passed, total coverage 67%
uv run ruff check ib_sec_mcp tests   # clean
uv run ruff format --check           # clean
```

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tests/analyzers/sentiment/test_technical.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        return self.df
+
+
+@pytest.fixture()
+def patch_ticker(monkeypatch: pytest.MonkeyPatch):
+    def _apply(df: pd.DataFrame) -> None:
+        monkeypatch.setattr(FakeTicker, "df", df)
+        monkeypatch.setattr("ib_sec_mcp.analyzers.sentiment.technical.yf.Ticker", FakeTicker)
+
+    return _apply
```
[the remark this answers is not in the document]
Comment (knishioka) [reply tests/analyzers/sentiment/test_technical.py]: Good catch — fixed in ce67b74. Moved `FakeTicker` inside the `patch_ticker` closure so the history DataFrame is captured per-call instead of via a shared class attribute, making it parallel-safe (pytest-xdist). Tests still green.
